service/k8sclient/k8sclient_test.go

```diff
@@ -39,6 +41,140 @@ import (
 	"github.com/percona-platform/dbaas-controller/utils/logger"
 )
 
+// VersionServiceClient represents a client for Version Service API.
+type VersionServiceClient struct {
+	url  string
+	http *http.Client
+}
+
+// NewVersionServiceClient creates a new client for given version service URL.
+func NewVersionServiceClient(url string) *VersionServiceClient {
+	return &VersionServiceClient{
+		url: url,
+		http: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+// componentsParams contains params to filter components in version service API.
+type componentsParams struct {
+	product        string
+	productVersion string
+}
+
+type matrix struct {
+	PXCOperator   map[string]interface{} `json:"pxcOperator,omitempty"`
+	PSMDBOperator map[string]interface{} `json:"psmdbOperator,omitempty"`
+}
+
+type Version struct {
+	Product        string `json:"product"`
+	ProductVersion string `json:"operator"`
+	Matrix         matrix `json:"matrix"`
+}
+
+// VersionServiceResponse represents response from version service API.
+type VersionServiceResponse struct {
+	Versions []Version `json:"versions"`
+}
+
+var errNoVersionsFound error = errors.New("no versions to compare current version with found")
```

Comment:
looks like errNoVersionsFound is not needed, because you are not handling this case in the other places and this error is private, so cannot be used outside of current package.

Reply:
> looks like errNoVersionsFound is not needed, because you are not handling this case in the other places and this error is private, so cannot be used outside of current package.

I return this error on two places. If I remove this I will return the errors.New("no versions to compare current version with found") there - I will repeat myself. I don't think that is better.